_posts/2020-08-26-bazel-3.5.md

```diff
@@ -0,0 +1,107 @@
+---
+layout: posts
+title: "Bazel 3.5"
+authors:
+  - aiuto
+---
+
+[Bazel 3.5](https://github.com/bazelbuild/bazel/releases/tag/3.5.0) has just been released.
+
+Bazel 3.5 is a minor release. It is an incremental update to Bazel 3.4.
+
+*Baseline*: 889bc0b523b47eeb38a72bf9bb6858ee525a7c7e
+
+*Cherry picks*:
+-   a7a0d48fbeb059ee60e77580e5d05baeefdd5699: Make no-op starlark transition not affect the output directory
+-   b37c51c7085f0aefe04034dd451acb847605ddb5: Add `include_prefix` and `strip_include_prefix` to `cc_common.compile`
+-   f6ad35fcde93f92c591778ed7db38d167f5bbc03: Delete `--experimental_transparent_compression`
+-   39bc97eab295bddb35b38bfc4a2ff3d2b15d034e: Remove `--experimental_action_args`
+-   b9706675a7abf6ceebb250f0b3dfa4087a0c35f6: Stop needlessly parsing WORKSPACE files from external repositories
+-   e574d558da17cfd0f818e7a937a07926aa270069: Allow hyphen char in `workspace` names
+-   9993785fa0c4fa4172aa31d306f3abea76833abf: Allow dot ('.') in `workspace` names
+
+## Incompatible changes
+
+- Remove the `--experimental_process_wrapper_wait_fix` flag.
+- Remove the `--experimental_ui_deduplicate` flag.
+- Remove the `--experimental_transparent_compression` flag.
+- Remove the `--experimental_action_args` flag.
+- Bazel now correctly prefers Xcode versions in `/Applications` over
+  Other paths, by default.  This resolves an issue with accidentally
+  picking up an Xcode version from a Time Machine backup or network
+  disk.  If you rely on the old behavior, move your desired Xcode
+  version to `/Applications`.
+
+## New features:
+
+- `cquery` now follows aspects when the `--include_aspects` flag is used.
+- Add support to `bazel/crosstool` for building arm64 on macos.
+
+## Important changes:
+
+- Fix resource shrinking when \<overlayable\> tags are used.
+- Remove legacy handling of `--extra_checks`
+- Remove old incompatible flag `--incompatible_symlinked_sandbox_expands_tree_artifacts_in_runfiles_tree`.
+- Update coverage configuration for Python, filegroup, and shell script
+  rules to distinguish between source and dependency attributes.
+
+- Add `InstrumentedFilesInfo` provider to Starlark globals.
+- Make filegroup always forward `InstrumentedFilesProvider` and not
+  collect any sources directly.
+- *Caution*: Update Starlark error reporting and the call stack,
+   which may cause regressions, such as errors that lack relevant
+   location information.
```

Comment:
Can we phrase this in a non-scary way?
> Starlark error reporting is updated. Please report regressions, such as errors that lack relevant location information.

Reply:
@michajlo How do you want to word this?

Reply:
This was from one of @alandonovan's changes.

Reply:
OK. After reading the comment in the PR, I don't think the caution is needed.
If we update the reporting so we get better call stack, error locations will need to change.
If people are depending on the exact error stack, their test is brittle.

Reply:
Sorry I didn't see this till now. When I wrote the CL I was concerned that there would be places in which error location information had gone missing, but there seem to have been no adverse effects reported on Blaze, and in many ways the errors are much improved. I concur that it is safe to strike the warning.

Reply:
> If we update the reporting so we get better call stack, error locations will need to change.
> If people are depending on the exact error stack, their test is brittle.

By the way, the analysis test mechanism (https://docs.bazel.build/versions/master/skylark/testing.html#failure-testing) positively invites users to write fragile tests of analysis failures, that depend on the precise wording of the Bazel error message _du jour_.